File: problem/Count_Submatrices_X_Y/numberOfSubmatrices.c
int numberOfSubmatrices(char **grid, int gridSize, int *gridColSize)
{
    int rows = gridSize;
    int cols = gridColSize[0];
    int *sumX = (int *)calloc(cols, sizeof(int));
    int *sumY = (int *)calloc(cols, sizeof(int));
    int res = 0;

    for (int i = 0; i < rows; i++)
    {
        int rx = 0, ry = 0;
        for (int j = 0; j < cols; j++)
        {
            if (grid[i][j] == 'X')
                rx++;
            else if (grid[i][j] == 'Y')
                ry++;
            sumX[j] += rx;
            sumY[j] += ry;
            if (sumX[j] > 0 && sumX[j] == sumY[j])
                res++;
        }
    }

    free(sumX);
    free(sumY);
    return res;
}

File: problem/Count_Submatrices_X_Y/numberOfSubmatrices.js
/**
 * @param {character[][]} grid
 * @return {number}
 */
var numberOfSubmatrices = function(grid) {
    const rows = grid.length;
    const cols = grid[0].length;
    const sumX = new Array(cols).fill(0);
    const sumY = new Array(cols).fill(0);
    let res = 0;

    for (let i = 0; i < rows; i++) {
        let rx = 0, ry = 0;
        for (let j = 0; j < cols; j++) {
            if (grid[i][j] === 'X') rx++;
            else if (grid[i][j] === 'Y') ry++;
            sumX[j] += rx;
            sumY[j] += ry;
            if (sumX[j] > 0 && sumX[j] === sumY[j]) res++;
        }
    }

    return res;
};